Scripts/Data/Type/Transformation/CollectionAggregator.cs

```diff
@@ -98,8 +88,29 @@ protected override TOutput Process(TInput input)
                 return default(TOutput);
             }
 
-            SetElementAtCurrentIndex(input);
+            SetElement(input);
             return ProcessCollection();
         }
+
+        /// <summary>
+        /// Preprocesses the value to be set in <see cref="CurrentIndex"/>
+        /// </summary>
+        /// <param name="value">The value to process.</param>
+        /// <returns>The processed value</returns>
+        protected virtual int PreprocessCurrentIndex(int value)
```

Comment:
It looks like this method can be removed and the property above just uses `ProcessIndex` like everything else.

Reply:
This is in anticipation of the code weaving stuff.

The `PreprocessCurrentIndex` will be the method that's auto woven into the `CurrentIndex` field.

But the actual function being done inside `PreprocessCurrentIndex` is used elsewhere so it's separated out.